src/app/store/state/login.state.ts

```diff
@@ -0,0 +1,13 @@
+import { User } from "src/app/entities/user";
+
+export interface ILoginState {
+    user: User | null;
+    error: string | null,
+    status: 'pending'|'loading'|'error'|'success'
```

Comment:
I would extract this into a separate type. Otherwise, in real projects it's more typical to see a boolean flag called loading.